### PR TITLE
DOP-1346 - MSEditor Gallery (Step 4/?)

### PR DESCRIPTION
**Pull request by andresmoschini**
Hi team!

In this PR I am making the gallery look like a gallery:

https://github.com/FromDoppler/doppler-editors-webapp/assets/1157864/8d6c839d-dca8-4f09-8b05-31c27667b995

@GusBaamonde, I am not sure about the structure and class names, and, of course about the styles. Maybe, we can fix the structure and class names in the review of this PR to make moving to the style guide easier.

### CODE REVIEW

File: src/components/custom-media-library/List.tsx [hunk 1]
```diff
@@ -18,18 +18,25 @@ export const List = ({
       TODO: show the list of images, allow to select and unselect them
     */}
     {isLoading ? <>Loading...</> : false}
-    <ul data-testid="image-list">
-      {images.map((x) => (
+    <ul className="gallery--showcase" data-testid="image-list">
```
Comment (andresmoschini) [inline src/components/custom-media-library/List.tsx:21]:
```suggestion
    <ul className="dp-image-gallery" data-testid="image-list">
```

File: src/components/custom-media-library/List.tsx [hunk 2]
```diff
@@ -18,18 +18,25 @@ export const List = ({
...
+      {images.map((x, i) => (
         <li key={x.name}>
-          <p>{JSON.stringify(x)}</p>
-          <p>
+          <label
+            className={`image__container ${
+              checkedImages.has(x) ? "checked" : "unchecked"
+            }`}
```
Comment (andresmoschini) [inline src/components/custom-media-library/List.tsx:27]:
```suggestion
            className="image__container"
```

File: src/components/custom-media-library/List.tsx [hunk 3]
```diff
@@ -18,18 +18,25 @@ export const List = ({
...
+            htmlFor={`image-item-${i}-check`}
+          >
+            <img src={x.thumbnailUrl150} alt={x.name} />
+            <div className="image__mask" />
```
Comment (andresmoschini) [inline src/components/custom-media-library/List.tsx:31]:
```suggestion
            <div className="dp-image-gallery--mask" />
```

File: src/components/custom-media-library/List.tsx [hunk 4]
```diff
@@ -18,18 +18,25 @@ export const List = ({
...
             <input
+              id={`image-item-${i}-check`}
               type="checkbox"
-              style={{ position: "relative" }}
               checked={checkedImages.has(x)}
               onChange={() => toggleCheckedImage(x)}
             />
-          </p>
+          </label>
+          <p className="image-name">{x.name}</p>
```
Comment (andresmoschini) [inline src/components/custom-media-library/List.tsx:39]:
```suggestion
          <p>{x.name}</p>
```

**Comment by andresmoschini** [comment]
@GusBaamonde I have updated the class names, could you review them again?

File: src/components/custom-media-library/Header.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ export const Header = ({ cancel }: { cancel: () => void }) => (
       name="close-modal"
       onClick={cancel}
     ></button>
-    <div className="gallery__header">
+    <div className="dp-image-gallery-header">
```
Comment (GusBaamonde) [inline src/components/custom-media-library/Header.tsx:11]: Si es un header, por que no usar un tag header en lugar de un div?

File: src/components/custom-media-library/List.tsx [hunk 6]
```diff
@@ -13,23 +13,28 @@ export const List = ({
   checkedImages: ReadonlySet<ImageItem>;
   toggleCheckedImage: (item: ImageItem) => void;
 }) => (
-  <div className="gallery__images">
+  <div className="dp-image-gallery-content">
```
Comment (GusBaamonde) [inline src/components/custom-media-library/List.tsx:16]: 👍

File: src/components/custom-media-library/List.tsx [hunk 7]
```diff
@@ -13,23 +13,28 @@ export const List = ({
...
+      {images.map((x, i) => (
         <li key={x.name}>
-          <p>{JSON.stringify(x)}</p>
-          <p>
+          <label
+            className="dp-image-gallery-thumbnail"
+            htmlFor={`image-item-${i}-check`}
+          >
+            <img src={x.thumbnailUrl150} alt={x.name} />
+            <div className="image__mask" />
```
Comment (GusBaamonde) [inline src/components/custom-media-library/List.tsx:29]: image__mask no cambiaba?

Comment (andresmoschini) [reply src/components/custom-media-library/List.tsx:29]: Si, debería, perdí el cambio, reviso...

File: src/components/custom-media-library/Header.tsx (hunk 5 above)
Comment (andresmoschini) [reply src/components/custom-media-library/Header.tsx:11]: Es el header de un modal, sería correcto usar `<header>` o solo tiene que haber uno por página?

File: src/components/custom-media-library/List.tsx (hunk 7 above)
Comment (andresmoschini) [reply src/components/custom-media-library/List.tsx]: Arregledo, gracias!

**Pull request merged**